odoo/addons/base/models/res_partner.py

```diff
@@ -762,9 +762,9 @@ def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_ui
         self = self.with_user(name_get_uid) if name_get_uid else self
         # as the implementation is in SQL, we force the recompute of fields if necessary
         self.recompute(['display_name'])
-        self.flush()
         if args is None:
             args = []
+        self._flush_search(args, fields=['display_name', 'email', 'ref', 'vat'])
```

Comment:
Careful, you don't take in account the `_get_name_search_order_by_fields` (override in account module) and the fields used in the methods should be also flushed.

Also, the `_flush_search` can be done only when you enter in the next `if`, which can also help.

Reply:
Hello,

Since the method from accounting returns a string, I will rebase this PR on master and make it return a list of fields as well.
For the flush, I kept it were it was, but I'll move it after the `check_access_rights`.

Thank you for the review.

Reply:
At the end, no need. In master, there's `_rec_names_search` which solves the issue. I'll close this PR.